torchx/mix.exs

```diff
@@ -56,8 +56,8 @@ defmodule Torchx.MixProject do
 
   defp deps do
     [
-      {:nx, "~> 0.4.0"},
-      # {:nx, path: "../nx"},
+      # {:nx, "~> 0.4.0"},
+      {:nx, path: "../nx"},
```

Comment:
We can still use `{:nx, "~> 0.4.0"}`, given the previous functionality is a superset. :)

Reply:
This PR also extends to accept rectangular identity shapes, so we can't test the updated implementations in EXLA and Torchx without pointing to latest Nx

Reply:
I'll defer the merge to you :)